nextcord/application_command.py

```
@@ -0,0 +1,883 @@
+        self.guild_id: Optional[int] = utils._get_as_snowflake(payload, "guild_id")
+        self.name: str = payload["name"]
+        self.description: str = payload["description"]
+        self.options: List[
+            ApplicationCommandResponseOption
+        ] = ApplicationCommandResponseOption._create_options(payload.get("options", []))
+        self.default_permission: Optional[bool] = payload.get(
+            "default_permission", True
+        )
+
+    @property
+    def guild(self) -> Optional[Guild]:
+        """Optional[:class:`Guild`]: Returns the :class:`Guild` associated with this Response, if any."""
+        return self._state._get_guild(self.guild_id)
+
+    @property
+    def signature(self) -> Tuple[str, int, Optional[int]]:
+        """Returns a simple high level signature of the command. No commands registered in the bot at the same time
+        should have identical signatures.
+
+        Returns
+        -------
+        name: :class:`str`
+            Name of the Application Command.
+        type: :class:`int`
+            Discord's integer value of the Application Command type
+        guild_id: Optional[:class:`int`]
+            The Guild ID associated with the Application Command. If None, it's a global command.
+        """
+        return self.name, self.type.value, self.guild_id
+
+
+class ApplicationCommandResponseOptionChoice:
+    """Represents a single choice in a list of options.
+
+    Attributes
+    ----------
+    name: :class:`str`
+        Name of the choice, this is what users see in Discord.
+    value: Union[:class:`str`, :class:`int`, :class:`float`]
+        Value of the choice, this is what Discord sends back to us.
+    """
+
+    def __init__(self, payload: Optional[dict] = None):
+        if not payload:
+            payload = {}
+        self.name: str = payload.get('name')
+        self.value: Union[str, int, float] = payload.get('value')
```

Given the above two *need* to exist, or appear to based on this. Wouldn't it be better to index with `[]` and let the keyerror be raised, or raise a custom exception if they don't exist. ``get`` will default to ``None`` afaik

---

yeah, if intended to none add the Optional[] type